python/Ganga/test/GPI/FileTests/TestMassStorageClient.py

```diff
@@ -30,13 +32,19 @@ def setUp(self):
         extra_opts=[('PollThread', 'autostart', 'False'),
                     ('Local', 'remove_workdir', 'False'),
                     ('TestingFramework', 'AutoCleanup', 'False'),
-                    ('Output', 'MassStorageFile', TestMassStorageClient.MassStorageTestConfig),
+                    ('Output', TestMassStorageClient.fileName, TestMassStorageClient.MassStorageTestConfig),
                     ('Output', 'FailJobIfNoOutputMatched', 'True')]
         super(TestMassStorageClient, self).setUp(extra_opts=extra_opts)
 
+    @classmethod
+    def getFileObject(cls):
+        """ Return an instance of the file we're wanting to test """
+        import Ganga.GPI as gpi
+        return getattr(gpi, cls.fileName)
```

Comment:
If you need a proxied version then using:
```python
fileClass = addProxy(MassStorageFile)
```
should suffice.

As for the plugin system, I used to feel the same but I actually think it's got a lot of use and power. It helps abstract away the interface and reduces the dependencies between classes.